snap/gadget.go

```diff
@@ -69,6 +72,71 @@ type VolumeContent struct {
 	Unpack bool `yaml:"unpack"`
 }
 
+// GadgetConnect describes an interface connection requested by the gadget
+// between seeded snaps. The syntax looks like this:
+//
+// [<plug-snap-id1>|system]:plug [[<slot-snap-id>|system]:slot]
+//
+// "system" or omitting the snap-id indicates a system plug or slot.
+// Fully omitting the slot part indicates a system slot with the same name
+// as the plug.
+type GadgetConnect struct {
+	PlugSnapID string
+	Plug       string
+	SlotSnapID string
+	Slot       string
+}
+
+func parseSnapIDColonName(s string) (snapID, name string, err error) {
+	parts := strings.Split(s, ":")
+	if len(parts) == 2 {
```

Comment:
You could squeeze this on one line: `if parts := strings.Split(s, ":"); len(parts) == 2 {`

Reply:
Doesn't mean it's a good idea, right? :)  The code looks very clean as is.